scala_config.bzl

```diff
@@ -28,7 +28,7 @@ def _store_config(repository_ctx):
     )
 
     # All versions supported
-    scala_versions = [scala_version]
+    scala_versions = [scala_version] + [v for v in repository_ctx.attr.scala_versions if v != scala_version]
```

Comment:
Hi, @aszady, all looks good to me. Just one question. Shouldn't there be a check that `scala_version` (default) is specified in `scala_versions` (all supported)?
Now `scala_config(scala_version = "2.12.18", scala_versions = ["3.3.1"])` would result in a default version `"2.12.18"` and all configured `["2.12.18", "3.3.1"]` I think this is different from what doc says. What do you think? Am I missing something?

Reply:
The doc currently says only about the variables `SCALA_VERSION`, `SCALA_VERSIONS` in the generated file `@io_bazel_rules_scala_config//:config.bzl`.

We can of course make them more consistent with the args to `scala_config` if it is too confusing.
I pushed a revision that enforces containing `scala_version` in `scala_versions` – instead of silently fixing the user's oversight.
One can also still not provide the `scala_versions` at all and it will default to `[scala_version]` (compatibility).

Reply:
Right that use-case without `scala_versions` :) somehow managed to forget.